testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -552,6 +554,260 @@ public void testJoinBaseTableWithView()
         assertAccessAllowed(viewOwnerSession, "DROP VIEW " + viewName);
     }
 
+    @Test
+    public void testAllowCallFunction()
+    {
+        reset();
+
+        String functionOwner = "function_owner";
+        CatalogSchemaFunctionName outerFunction = new CatalogSchemaFunctionName("memory", new SchemaFunctionName("default", "function_allow_outer"));
+        CatalogSchemaFunctionName innerFunction = new CatalogSchemaFunctionName("memory", new SchemaFunctionName("default", "function_allow_inner"));
+
+        Session functionOwnerSession = TestingSession.testSessionBuilder()
+                .setIdentity(Identity.forUser(functionOwner)
+                        .build())
+                .setCatalog(getSession().getCatalog())
+                .setSchema(getSession().getSchema())
+                .build();
+
+        // simply create a function
+        assertAccessAllowed(
+                functionOwnerSession,
+                "CREATE FUNCTION memory.default.function_allow_inner (x integer) RETURNS bigint RETURN x + 42");
+        assertThat(systemSecurityMetadata.getFunctionOwner(innerFunction)).isEqualTo(functionOwner);
+
+        // assert that function can be called for both definer session and default session
+        assertAccessAllowed(
+                functionOwnerSession,
+                "SELECT memory.default.function_allow_inner(2)");
+        assertAccessAllowed("SELECT memory.default.function_allow_inner(2)");
+
+        // simply create another function, internally calls the first function
+        assertAccessAllowed(
+                functionOwnerSession,
+                "CREATE FUNCTION memory.default.function_allow_outer (x integer) RETURNS bigint RETURN x + memory.default.function_allow_inner(58)");
+        assertThat(systemSecurityMetadata.getFunctionOwner(outerFunction)).isEqualTo(functionOwner);
+
+        // assert that THE outer function can be called for both definer session and default session
+        assertAccessAllowed(
+                functionOwnerSession,
+                "SELECT memory.default.function_allow_outer(2)",
+                privilege(getSession().getUser(), "memory.default.function_allow_inner", EXECUTE_FUNCTION));
+        assertAccessAllowed("SELECT memory.default.function_allow_outer(2)");
+
+        // assert that lack of privileges to execute inner function doesn't block calling it through outer one
+        assertAccessAllowed(
+                "SELECT memory.default.function_allow_outer(2)",
+                privilege(getSession().getUser(), "memory.default.function_allow_inner", EXECUTE_FUNCTION));
+    }
+
+    @Test
+    public void testDenyCallFunction()
```

Comment:
I'm not sure this is what @kokosing had in mind, but this name is not right. It's more about roles than denying.